core/arch/x86_64/include/kernel/thread.h

```diff
@@ -78,167 +64,11 @@ struct thread_smc_args {
-	uint32_t usr_lr;
-	uint32_t pad;
-	uint32_t spsr;
-	uint32_t elr;
-	uint32_t r0;
-	uint32_t r1;
-	uint32_t r2;
-	uint32_t r3;
-	uint32_t r4;
-	uint32_t r5;
-	uint32_t r6;
-	uint32_t r7;
-	uint32_t r8;
-	uint32_t r9;
-	uint32_t r10;
-	uint32_t r11;
-	uint32_t ip;
-};
-#endif /*ARM32*/
-
-#ifdef ARM64
-struct thread_abort_regs {
-	uint64_t x0;	/* r0_usr */
-	uint64_t x1;	/* r1_usr */
-	uint64_t x2;	/* r2_usr */
-	uint64_t x3;	/* r3_usr */
-	uint64_t x4;	/* r4_usr */
-	uint64_t x5;	/* r5_usr */
-	uint64_t x6;	/* r6_usr */
-	uint64_t x7;	/* r7_usr */
-	uint64_t x8;	/* r8_usr */
-	uint64_t x9;	/* r9_usr */
-	uint64_t x10;	/* r10_usr */
-	uint64_t x11;	/* r11_usr */
-	uint64_t x12;	/* r12_usr */
-	uint64_t x13;	/* r13/sp_usr */
-	uint64_t x14;	/* r14/lr_usr */
-	uint64_t x15;
-	uint64_t x16;
-	uint64_t x17;
-	uint64_t x18;
-	uint64_t x19;
-	uint64_t x20;
-	uint64_t x21;
-	uint64_t x22;
-	uint64_t x23;
-	uint64_t x24;
-	uint64_t x25;
-	uint64_t x26;
-	uint64_t x27;
-	uint64_t x28;
-	uint64_t x29;
-	uint64_t x30;
-	uint64_t elr;
-	uint64_t spsr;
-	uint64_t sp_el0;
-};
-#endif /*ARM64*/
-
-#ifdef ARM32
-struct thread_svc_regs {
-	uint32_t spsr;
-	uint32_t r0;
-	uint32_t r1;
-	uint32_t r2;
-	uint32_t r3;
-	uint32_t r4;
-	uint32_t r5;
-	uint32_t r6;
-	uint32_t r7;
-	uint32_t lr;
-};
-#endif /*ARM32*/
-#ifdef ARM64
-struct thread_svc_regs {
-	uint64_t elr;
-	uint64_t spsr;
-	uint64_t x0;	/* r0_usr */
-	uint64_t x1;	/* r1_usr */
-	uint64_t x2;	/* r2_usr */
-	uint64_t x3;	/* r3_usr */
-	uint64_t x4;	/* r4_usr */
-	uint64_t x5;	/* r5_usr */
-	uint64_t x6;	/* r6_usr */
-	uint64_t x7;	/* r7_usr */
-	uint64_t x8;	/* r8_usr */
-	uint64_t x9;	/* r9_usr */
-	uint64_t x10;	/* r10_usr */
-	uint64_t x11;	/* r11_usr */
-	uint64_t x12;	/* r12_usr */
-	uint64_t x13;	/* r13/sp_usr */
-	uint64_t x14;	/* r14/lr_usr */
-	uint64_t x30;
-	uint64_t sp_el0;
-#ifdef CFG_SECURE_PARTITION
-	uint64_t x15;
-	uint64_t x16;
-	uint64_t x17;
-	uint64_t x18;
-	uint64_t x19;
-	uint64_t x20;
-	uint64_t x21;
-	uint64_t x22;
-	uint64_t x23;
-	uint64_t x24;
-	uint64_t x25;
-	uint64_t x26;
-	uint64_t x27;
-	uint64_t x28;
-	uint64_t x29;
-#endif
-	uint64_t pad;
-} __aligned(16);
-#endif /*ARM64*/
-
-#ifdef ARM32
-struct thread_ctx_regs {
-	uint32_t r0;
-	uint32_t r1;
-	uint32_t r2;
-	uint32_t r3;
-	uint32_t r4;
-	uint32_t r5;
-	uint32_t r6;
-	uint32_t r7;
-	uint32_t r8;
-	uint32_t r9;
-	uint32_t r10;
-	uint32_t r11;
-	uint32_t r12;
-	uint32_t usr_sp;
-	uint32_t usr_lr;
-	uint32_t svc_spsr;
-	uint32_t svc_sp;
-	uint32_t svc_lr;
-	uint32_t pc;
-	uint32_t cpsr;
-};
-#endif /*ARM32*/
-
-struct thread_ctx_regs {
-	uint64_t rflags;
-	uint64_t rip;
-	uint64_t r10;
-	uint64_t r11;
-	uint64_t r12;
-	uint64_t r13;
-	uint64_t r14;
-	uint64_t r15;
-};
 
 struct thread_specific_data {
 	TAILQ_HEAD(, ts_session) sess_stack;
 	struct ts_ctx *ctx;
 	struct pgt_cache pgt_cache;
-#ifdef CFG_CORE_FFA
-	uint32_t rpc_target_info;
-#endif
 	uint32_t abort_type;
 	uint32_t abort_descr;
 	vaddr_t abort_va;
```

Comment:
Since you have removed abort.c, do we still need these abort related fields struct thread_specific_data?

Reply:
resolved